File: eRxWeb/SPA/Model/message.model.ts
export class MessageModel {
    Message: string;
    Icon: MessageIcon;
    Tag: string;
    ShowCloseButton: boolean;

    public constructor(message: string, icon: MessageIcon, tag: string, showclosebutton: boolean )
    {
        this.Message = message;
        this.Tag = tag;
        this.Icon = icon;
        this.ShowCloseButton = showclosebutton;
    }
}

export enum MessageIcon {
    Success = 0,
    Error,
    Information
}

File: eRxWeb/ServerModel/MessageModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace eRxWeb.ServerModel
{
    public enum MessageIcon
    {
        Success = 0,
        Error=1,
        Information=2
    }
    public class MessageModel
    {
        public MessageModel()
        {
            ShowCloseButton = true;
        }

        public MessageModel(string message, MessageIcon icon = MessageIcon.Success)
        {
            Message = message;
            Icon = icon;
            ShowCloseButton = true;
        }

        public MessageModel(string message, MessageIcon icon, bool showCloseButton)
        {
            Message = message;
            Icon = icon;
            ShowCloseButton = showCloseButton;
        }

        public string Message { get; set; }
        public MessageIcon Icon { get; set; }
        public string Tag { get; set; }
        public bool ShowCloseButton { get; set; }
    }
   
}